packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -42,7 +42,9 @@ export default createRule({
       fixer: TSESLint.RuleFixer,
       node: TSESTree.SwitchStatement,
       missingBranchTypes: Array<ts.Type>,
+      symbolName?: string,
     ): TSESLint.RuleFix | null {
+      const identifierRegex = /^[a-zA-Z_$][0-9a-zA-Z_$]*$/;
```

Comment:
The problem we've got is that `a-zA-Z` doesn't include all of the "letter-like" characters that are actually allowed in the chinese/japanese/korean/cyrillic/etc characters, accented characters, etc. So for a large number of codebases this regex will actually cause a false match. A false match isn't the end of the world, but it's going to be annoying for non-english speakers.

This is the actual regex of allowed characters within a variable name: https://github.com/estools/esutils/blob/a825f91fd1d1e3a9fff84227cb06c011d8a0b9e8/lib/code.js#L39-L44

It's a mess.

TypeScript needs to parse identifiers itself, and thankfully they export two functions that we can use here: `isIdentifierStart` and `isIdentifierPart`.

You can use this function instead to determine if the name requires quoting:
```ts
const compilerOptions = service.program.getCompilerOptions();
function requiresQuoting(name: string): boolean {
  if (name.length === 0) {
    return true;
  }
  if (!ts.isIdentifierStart(name.charCodeAt(0), compilerOptions.target)) {
    return true;
  }

  for (let i = 1; i < name.length; i += 1) {
    if (!ts.isIdentifierPart(name.charCodeAt(i), compilerOptions.target)) {
      return true;
    }
  }

  return false;
}
```

Reply:
Thank you! Will make the changes as soon as I can find some time.